src/routes/messages/mappers/message-mapper.ts

```diff
@@ -16,6 +17,32 @@ export class MessageMapper {
     private readonly addressInfoHelper: AddressInfoHelper,
   ) {}
 
+  async mapMessageItems(
+    chainId: string,
+    domainMessages: DomainMessage[],
+    safe: Safe,
+  ): Promise<MessageItem[]> {
+    return Promise.all(
+      domainMessages.map(async (domainMessage) => {
+        const message = await this.mapMessage(chainId, domainMessage, safe);
+        return new MessageItem(
+          message.messageHash,
+          message.status,
+          message.logoUri,
+          message.name,
+          message.message,
+          message.creationTimestamp,
+          message.modifiedTimestamp,
+          message.confirmationsSubmitted,
+          message.confirmationsRequired,
+          message.proposedBy,
+          message.confirmations,
+          message.preparedSignature,
+        );
```

Comment:
This could be simplified if we merge both `MessageItem` and `Message` entities. See: https://github.com/5afe/safe-client-gateway-nest/pull/279/files#r1118446395 🙂